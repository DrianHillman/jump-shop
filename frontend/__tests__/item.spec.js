import ItemComponent from '../components/Item';
import { shallow } from 'enzyme';

const fakeItem = {
  id: 'ABC123',
  title: 'A Cool Item',
  price: 5000,
  description: 'This item is reall cool!',
  image: 'dog.jpg',
  largeImage: 'large-dog.jpg',
};

describe('<Item />', () => {
  it('should render the price tag and title properly', () => {
    const wrapper = shallow(<ItemComponent item={fakeItem} />);
    const PriceTag = wrapper.find('PriceTag');
    const Title = wrapper.find('Title a');

    // console.log(wrapper.debug());
    // console.log(PriceTag.children().debug);

    expect(PriceTag.children().text()).toBe('$50');
    expect(Title.text()).toBe(fakeItem.title);
  });

  it('should render the image properly', () => {
    const wrapper = shallow(<ItemComponent item={fakeItem} />);
    const img = wrapper.find('img');

    expect(img.props().src).toBe(fakeItem.image);
    expect(img.props().alt).toBe(fakeItem.title);
  });

  it('should render the buttons properly', () => {
    const wrapper = shallow(<ItemComponent item={fakeItem} />);
    const buttonList = wrapper.find('.buttonList');

    console.log(wrapper.debug());

    expect(buttonList.children()).toHaveLength(3);
    expect(buttonList.find('Link').exists()).toBe(true);
    expect(buttonList.find('AddToCart').exists()).toBe(true);
    expect(buttonList.find('DeleteButton').exists()).toBe(true);
  });
});
