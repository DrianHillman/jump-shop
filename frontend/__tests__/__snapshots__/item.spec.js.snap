// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Item /> should render and match the snapshot 1`] = `
<ItemStyles__Item>
  <img
    alt="A Cool Item"
    src="dog.jpg"
  />
  <Title>
    <Link
      href={
        Object {
          "pathname": "/item",
          "query": Object {
            "id": "ABC123",
          },
        }
      }
    >
      <a>
        A Cool Item
      </a>
    </Link>
  </Title>
  <PriceTag>
    $50
  </PriceTag>
  <p>
    This item is reall cool!
  </p>
  <div
    className="buttonList"
  >
    <Link
      href={
        Object {
          "pathname": "/update",
          "query": Object {
            "id": "ABC123",
          },
        }
      }
    >
      <button>
        Edit
         
        <span
          aria-label="Pencil Emoji"
          aria-labelledby="Edit"
          role="img"
          style={
            Object {
              "padding": "0 0.5rem",
            }
          }
        >
          ✏️
        </span>
      </button>
    </Link>
    <AddToCart
      id="ABC123"
    />
    <DeleteButton
      id="ABC123"
    >
      Delete Item
    </DeleteButton>
  </div>
</ItemStyles__Item>
`;
